tests/components/mqtt/test_common.py

```diff
@@ -527,32 +523,38 @@ async def help_test_discovery_update(
     async_fire_mqtt_message(hass, f"homeassistant/{domain}/bla/config", discovery_data1)
     await hass.async_block_till_done()
 
-    if state_data1:
-        for (topic, data) in state_data1:
-            async_fire_mqtt_message(hass, topic, data)
     state = hass.states.get(f"{domain}.beer")
     assert state is not None
     assert state.name == "Beer"
-    if state1:
-        assert state.state == state1
-    if attributes1:
-        for (attr, value) in attributes1:
-            assert state.attributes.get(attr) == value
+
+    if state_data1:
+        for (mqtt_messages, expected_state, attributes) in state_data1:
+            for (topic, data) in mqtt_messages:
+                async_fire_mqtt_message(hass, topic, data)
+            state = hass.states.get(f"{domain}.beer")
+            if expected_state:
+                assert state.state == expected_state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)